src/qutip_qip/device/circuitqed.py

```diff
@@ -27,97 +31,145 @@ class SCQubits(ModelProcessor):
     Parameters
     ----------
     num_qubits: int
-        Number of qubits
-    t1, t2: float or list, optional
-        Coherence time for all qubit or each qubit
+        The number of qubits in the system.
+    dims: list, optional
+        The dimension of each component system.
+        Default value is a qubit system of ``dim=[2,2,2,...,2]``.
     zz_crosstalk: bool, optional
-        if zz cross talk is included.
+        If ZZ cross-talk is included.
     **params:
-        Keyword argument for hardware parameters, in the unit of GHz.
-        Each can should be given as list:
-
-        - ``wq``: qubit bare frequency, default 5.15 and 5.09
-          for each pair of superconducting qubits,
-          e.g. ``[5.15, 5.09, 5.15, ...]``
-        - ``wr``: resonator bare frequency, default ``[5.96]*num_qubits``
-        - ``g``: The coupling strength between the resonator and the qubits,
-          default ``[0.1]*(num_qubits - 1)``
-        - ``alpha``: anharmonicity for each superconducting qubit,
-          default ``[-0.3]*num_qubits``
-        - ``omega_single``: control strength for single-qubit gate,
-          default ``[0.01]*num_qubits``
-        - ``omega_cr``: control strength for cross resonance gate,
-          default ``[0.01]*num_qubits``
-
-    Attributes
+        Hardware parameters. See :obj:`SCQubitsModel`.
+    """
+
+    def __init__(self, num_qubits, dims=None, zz_crosstalk=False, **params):
+        if dims is None:
+            dims = [3] * num_qubits
+        model = SCQubitsModel(
+            num_qubits=num_qubits,
+            dims=dims,
+            zz_crosstalk=zz_crosstalk,
+            **params,
+        )
+        super(SCQubits, self).__init__(model=model)
+        self.native_gates = ["RX", "RY", "CNOT"]
+        self._default_compiler = SCQubitsCompiler
+        self.pulse_mode = "continuous"
+
+    def topology_map(self, qc):
+        return to_chain_structure(qc)
+
+
+class SCQubitsModel(Model):
+    """
+    The physical model for superconducting qubits with fixed frequency.
```

Comment:
```suggestion
    The physical model for superconducting-qubit model processor with fixed frequency.
```

Reply:
We should add briefly a bit more details about the Hamiltonian. I propose we write it using the math directive in latex.